include/sparrow/fixed_size_layout.hpp

```diff
@@ -0,0 +1,369 @@
+        void advance(difference_type n);
+        difference_type distance_to(const self_type& hs) const;
+        bool equal(const self_type& rhs) const;
+        bool less_than(const self_type& rhs) const;
+
+        pointer m_pointer = nullptr;
+
+        friend class iterator_access;
+    };
+
+    /**
+     * A contiguous layout for fixed size types.
+     *
+     * This class provides a contiguous layout for fixed size types, such as `uint8_t`, `int32_t`, etc.
+     * It iterates over the first buffer in the array_data, and uses the bitmap to skip over null.
+     * The bitmap is assumed to be present in the array_data.
+     *
+     * @tparam T The type of the elements in the layout's data buffer.
+     *           A fixed size type, such as a primitive type.
+     *
+     * @note This class is not thread-safe, exception-safe, copyable, movable, equality comparable.
+     */
+    template <class T>
+    class fixed_size_layout
+    {
+    public:
+
+        using self_type = fixed_size_layout<T>;
+        using inner_value_type = T;
+        using inner_reference = inner_value_type&;
+        using inner_const_reference = const inner_reference;
+        using bitmap_type = array_data::bitmap_type;
+        using bitmap_reference = typename bitmap_type::reference;
+        using bitmap_const_reference = typename bitmap_type::const_reference;
+        using value_type = std::optional<inner_value_type>;
+        using reference = reference_proxy<self_type>;
+        using const_reference = const_reference_proxy<self_type>;
+        using pointer = inner_value_type*;
+        using const_pointer = const inner_value_type*;
+        using size_type = std::size_t;
+        using difference_type = std::ptrdiff_t;
+
+        using const_bitmap_iterator = array_data::bitmap_type::const_iterator;
+        using const_value_iterator = fixed_size_layout_value_iterator<T, true>;
+
+        using bitmap_iterator = array_data::bitmap_type::iterator;
+        using value_iterator = fixed_size_layout_value_iterator<T, false>;
+
+        using const_bitmap_range = std::ranges::subrange<const_bitmap_iterator>;
+        using const_value_range = std::ranges::subrange<const_value_iterator>;
+
+        using bitmap_range = std::ranges::subrange<bitmap_iterator>;
+        using value_range = std::ranges::subrange<value_iterator>;
+
+        // TODO: implement with `begin` and `end` once the iterator is available.
+        // using iterator = layout_iterator<self_type, false>;
+        // using const_iterator = layout_iterator<self_type, true>;
+
+        explicit fixed_size_layout(array_data p);
+
+        size_type size() const;
+
+        reference operator[](size_type i);
+        const_reference operator[](size_type i) const;
+
+        bitmap_range bitmap();
+        value_range values();
+
+        const_bitmap_range bitmap() const;
+        const_value_range values() const;
+
+    private:
+        // We only use the first buffer and the bitmap.
```

Comment:
Not sure about this comment, it might not be the best place to document this, it's kind of out of context... (thinking about it)